app/js/actions/artifacts.js

```diff
@@ -119,7 +120,8 @@ export const refreshArtifacts = () => {
         .then((json) => {
             json.artifacts.forEach(artifact => dispatch(newArtifact(artifact)));
         })
-        .then(() => dispatch(actions.checkTypes()));
+        .then(() => dispatch(actions.checkTypes()))
+        .then(() => dispatch(actions.checkImportableTypes()));
```

Comment:
I think you could probably move this to `connection.js:establishConnection` instead. Since the types that are importable really aren't going to change between sessions.